### PR TITLE
Implement Idea Hub widget support for creating idea draft posts (3357).

### PR DESCRIPTION
**Pull request by johnPhillips**
## Summary

Addresses issue #3357

## Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [ ] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js [hunk 1]
```diff
@@ -61,9 +68,12 @@ const Idea = ( { postEditURL, name, text, topics, buttons } ) => {
 	}, [ name ] );
 
 	const handleCreate = useCallback( () => {
+		setIsProcessing( true );
+		createIdeaDraftPost( { name, text, topics } );
 		// @TODO: Implement callback.
 		global.console.log( `Created: ${ name }` );
-	}, [ name ] );
+		setIsProcessing( false );
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js:75]: This `@TODO` and the following `console.log` aren't needed anymore.

File: assets/js/modules/idea-hub/datastore/draft-ideas.js [hunk 2]
```diff
@@ -38,6 +38,12 @@ const fetchCreateIdeaDraftPostStore = createFetchStore( {
 		return {
 			...state,
 			draftPostIdeas: [ ...( state.draftPostIdeas || [] ), ideaDraftPost ],
+			newIdeas: [
+				...( state.newIdeas || [] ),
+			].filter( ( { name } ) => name !== ideaDraftPost.name ),
+			savedIdeas: [
+				...( state.savedIdeas || [] ),
+			].filter( ( { name } ) => name !== ideaDraftPost.name ),
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/datastore/draft-ideas.js:46]: No need to spread `newIdeas` and `saveIdeas`. You can call the `filter` function directly:
```suggestion
			newIdeas: ( state.newIdeas || [] ).filter( ( { name } ) => name !== ideaDraftPost.name ),
			savedIdeas: ( state.savedIdeas || [] ).filter( ( { name } ) => name !== ideaDraftPost.name ),
```

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
Thanks, @johnPhillips. It almost looks good to me. Added a few minor comments. Once it's fixed, it should be good to go.

File: assets/js/modules/idea-hub/datastore/draft-ideas.js (hunk 2 above)
Comment (johnPhillips) [reply assets/js/modules/idea-hub/datastore/draft-ideas.js:46]: Good point, thanks

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js [hunk 3]
```diff
@@ -61,9 +68,10 @@ const Idea = ( { postEditURL, name, text, topics, buttons } ) => {
 	}, [ name ] );
 
 	const handleCreate = useCallback( () => {
-		// @TODO: Implement callback.
-		global.console.log( `Created: ${ name }` );
-	}, [ name ] );
+		setIsProcessing( true );
+		createIdeaDraftPost( { name, text, topics } );
```
Comment (eugene-manuilov) [inline assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js:72]: @johnPhillips we also need to wait till the new draft post is created.

**Review by eugene-manuilov — CHANGES_REQUESTED** [review]
One more thing.

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js (hunk 3 above)
Comment (johnPhillips) [reply assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js]: I assume you mean adding an `await` here?

**Review by eugene-manuilov — APPROVED** [review]
LGTM!

File: assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js (hunk 3 above)
Comment (eugene-manuilov) [reply assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js]: Yep.

**Pull request merged**